collect_app/src/main/res/layout/main_menu_button.xml

```diff
@@ -0,0 +1,53 @@
+<?xml version="1.0" encoding="utf-8"?>
+<androidx.constraintlayout.widget.ConstraintLayout
```

Comment:
There's a lot of "one-off" styling here. I'd really like to look at creating these buttons using standard dimensions and theme attributes as much as possible.

To me, this component is essentially a rebuild of the `MaterialButton` (with Material 3 styles) that adds a "trailing view". Given that, I think we want to draw as much from the [Material 3 Button specs](https://m3.material.io/components/buttons/specs) and the corresponding [Material Components docs](https://github.com/material-components/material-components-android/blob/master/docs/components/Button.md) as possible. That'll mean that any changes to the theme can be reflected in the buttons without needing to go back in and tweak. 

For instance, here's a rework using Material 3 theme attributes and our standard dimensions:

```xml
<?xml version="1.0" encoding="utf-8"?>
<androidx.constraintlayout.widget.ConstraintLayout xmlns:android="http://schemas.android.com/apk/res/android"
    xmlns:app="http://schemas.android.com/apk/res-auto"
    xmlns:tools="http://schemas.android.com/tools"
    android:layout_width="match_parent"
    android:layout_height="wrap_content"
    android:layout_margin="@dimen/margin_extra_small"
    android:background="@drawable/main_menu_button_background"
    android:elevation="4dp">

    <ImageView
        android:id="@+id/icon"
        android:layout_width="wrap_content"
        android:layout_height="wrap_content"
        android:layout_marginStart="@dimen/margin_extra_large"
        android:layout_marginTop="@dimen/margin"
        android:layout_marginBottom="@dimen/margin"
        app:layout_constraintBottom_toBottomOf="parent"
        app:layout_constraintStart_toStartOf="parent"
        app:layout_constraintTop_toTopOf="parent"
        tools:src="@drawable/ic_delete" />

    <TextView
        android:id="@+id/name"
        android:layout_width="wrap_content"
        android:layout_height="wrap_content"
        android:layout_marginStart="@dimen/margin"
        android:ellipsize="end"
        android:maxLines="1"
        android:textAppearance="?textAppearanceLabelLarge"
        app:layout_constraintBottom_toBottomOf="@id/icon"
        app:layout_constraintStart_toEndOf="@id/icon"
        app:layout_constraintTop_toTopOf="@id/icon"
        tools:text="Fill Blank Form" />

    <TextView
        android:id="@+id/number"
        android:layout_width="wrap_content"
        android:layout_height="wrap_content"
        android:layout_marginEnd="@dimen/margin_extra_large"
        android:textAppearance="?textAppearanceLabelLarge"
        app:layout_constraintBottom_toBottomOf="@id/name"
        app:layout_constraintEnd_toEndOf="parent"
        app:layout_constraintTop_toTopOf="@id/name"
        tools:text="10" />

</androidx.constraintlayout.widget.ConstraintLayout>
```

And the background:

```xml
<?xml version="1.0" encoding="utf-8"?>
<ripple xmlns:android="http://schemas.android.com/apk/res/android"
    android:color="@color/color_on_surface_low_emphasis">
    <item>
        <shape android:shape="rectangle">
            <corners android:radius="100dp" />
            <solid android:color="?colorSurfaceContainer" />
        </shape>
    </item>
</ripple>
```

For the ripple color, I'm following Material Components' example of using the low emphasis (low alpha) version of `colorOnSurface` which we already have defined (a selector that uses the theme color and applies an alpha). One complication is that the button color seems like it should probably use the new ["Surface Container" color](https://m3.material.io/styles/color/the-color-system/tokens), but Material Components does not seem to have implemented that as an attribute. We can just define it ourselves in `attrs.xml`, and it seems to be like the `#f7f7f7` color we were using before is already the right fit - it should just be defined at a theme level rather than an outlying custom color. 

The resulting button looks like this:

<img width="410" alt="Screenshot 2023-04-18 at 14 21 26" src="https://user-images.githubusercontent.com/556280/232775699-7bfd2496-0691-4761-963e-e1174bc481ed.png">

It's not *exactly* like the mockup @alyblenkin created, but it's very close and I think we could add in a few tweaks (using a larger typography token for example) to get it to a place where everyone is happy.

In general, I think when working with these kinds of mockups we should get as close as we can with Material 3 components, styles and attributes and then look at tweaks we can make and if they're worth taking on the maintenance of custom styles.

Reply:
I thought about using our dimes but assumed that those specified on Figma should be used.

Reply:
We discussed this on a call and agreed that a good approach for using Figma mockups is just to treat them as a guide and then build the closest thing with Android's Material Components. Then we can do design review with @alyblenkin and discuss what might be important to tweak and if it is worth doing so. Does that make sense for you @alyblenkin? I've generally made the assumption that you're not treating the Figma designs as sacred, but it would be good to double-check.

Reply:
@grzesiek2010 @seadowg - Hey both, yes, the mockups are just a guide, not sacred! The Figma components I use are often a hybrid of M2 and M3 because they don't always have what we need, so they will never be pixel-perfect. Using the closest thing to Android's Material Components and then reviewing it sounds great! 

I usually follow this design process: 
- understand the user need and pain points
- mock something up to test the idea
- validate our assumptions by getting user feedback (internally/externally) 
- iterate and align with you guys to make sure it's doable 
- design review after you've built it to make sure the result is still achieving the user goal if we've had to deviate from the designs 
- test working software to get user feedback 

For me, the goal is to spend less time getting everything 100% accurate in Figma because the users never see it. I'm entirely on board with doing what works to get something usable into people's hands!

Reply:
@seadowg I have some questions about the layout you have reworked:
- why did you remove `layout_marginHorizontal` from the main (constraint) layout?
- why do you use `@dimen/margin` not our `@dimen/margin_standard`?
- why did you make the text smaller?
- now the text and the number use the same text size/style should I keep that? on Figma, the number text was bigger.

Reply:
Good questions.

> why did you remove layout_marginHorizontal from the main (constraint) layout?

I just went with the smallest margin we had around the whole view to make sure the elevation doesn't get cut off by margins a layout the view is placed in. It might be better to have different vertical and horizontal ones like you had before - it looks to me like `MaterialButton` has a larger vertical (internal) margin than horizontal.

> why do you use @dimen/margin not our @dimen/margin_standard?

Arggghhh that's a mistake I always make! `@dimens/margin` is always defined in some Android library (in this case it looks like it's the color picker we use). They happen to be the same (both 16dp),  but we should definitely use our one instead.

> why did you make the text smaller?

I used the typography attribute (`?textAppearanceLabelLarge`) that the [specs for Elevated use](https://m3.material.io/components/buttons/specs#7026d152-e4c1-4de9-9d06-5631c09c3f4b). There's examples of the Android specific implementation of these in [the Material Components docs](https://github.com/material-components/material-components-android/blob/master/docs/components/Button.md#anatomy-and-key-properties) as well. We could probably boost the size if it doesn't feel big enough, but I think is an example of where I think it makes sense to start with the default and then discuss if we need to tweak. The different standard typography attributes available to us are listed here: https://github.com/material-components/material-components-android/blob/master/docs/theming/Typography.md. We'll need to set up the attribute in `theme.xml` for any tokens we haven't already (as we've done for `textAppearanceLabelLarge`) as we're not using a Material 3 base theme yet.

> now the text and the number use the same text size/style should I keep that? on Figma, the number text was bigger.

It looked good with them as the same size to me, but I could definitely understand it might make sense to emphasize one (or boost the size of the number seeing as it's a small amount of text). Again we can just use a larger typography token for that. Definitely a question for @alyblenkin!

Reply:
@seadowg 

> Good questions.
> 
> > why did you remove layout_marginHorizontal from the main (constraint) layout?
> 
> I just went with the smallest margin we had around the whole view to make sure the elevation doesn't get cut off by margins a layout the view is placed in. It might be better to have different vertical and horizontal ones like you had before - it looks to me like `MaterialButton` has a larger vertical (internal) margin than horizontal.
> 
> > why do you use @dimen/margin not our @dimen/margin_standard?
> 
> Arggghhh that's a mistake I always make! `@dimens/margin` is always defined in some Android library (in this case it looks like it's the color picker we use). They happen to be the same (both 16dp), but we should definitely use our one instead.
> 
> > why did you make the text smaller?
> 
> I used the typography attribute (`?textAppearanceLabelLarge`) that the [specs for Elevated use](https://m3.material.io/components/buttons/specs#7026d152-e4c1-4de9-9d06-5631c09c3f4b). There's examples of the Android specific implementation of these in [the Material Components docs](https://github.com/material-components/material-components-android/blob/master/docs/components/Button.md#anatomy-and-key-properties) as well. We could probably boost the size if it doesn't feel big enough, but I think is an example of where I think it makes sense to start with the default and then discuss if we need to tweak. The different standard typography attributes available to us are listed here: https://github.com/material-components/material-components-android/blob/master/docs/theming/Typography.md. We'll need to set up the attribute in `theme.xml` for any tokens we haven't already (as we've done for `textAppearanceLabelLarge`) as we're not using a Material 3 base theme yet.
> 
> > now the text and the number use the same text size/style should I keep that? on Figma, the number text was bigger.
> 
> It looked good with them as the same size to me, but I could definitely understand it might make sense to emphasize one (or boost the size of the number seeing as it's a small amount of text). Again we can just use a larger typography token for that. Definitely a question for @alyblenkin!

In the figma the number text is slightly smaller (16px) than the label text (18px), which is typically the pattern we want to follow to emphasize the label. I'm not opposed to them being the same size because the label text is so short, although that might not be the case with other languages. Happy to go with the default and then test it to make sure it works well with longer label text!

Reply:
Looks great to me!